src/scripts/main.js

```diff
@@ -1,3 +1,163 @@
 'use strict';
 
-// write code here
+const thead = document.querySelector('thead');
+const tbody = document.querySelector('tbody');
+
+thead.addEventListener('click', (e) => {
+  const target = e.target;
+
+  if (target.tagName === 'TH') {
+    const index = target.cellIndex;
+
+    sortTable(index);
+  }
+});
+
+function sortTable(indexColumn) {
+  const column = indexColumn;
+  const rows = Array.from(tbody.querySelectorAll('tr'));
+
+  rows.sort((a, b) => {
+    if (column === 4) {
+      const aValue = a.cells[column].innerText.replace(/[$,]/g, '');
+      const bValue = b.cells[column].innerText.replace(/[$,]/g, '');
+
+      return aValue - bValue;
+    }
+
+    if (column === 3) {
+      return a.cells[column].innerText - b.cells[column].innerText;
+    }
+
+    const aText = a.cells[column].innerText.toLowerCase();
+    const bText = b.cells[column].innerText.toLowerCase();
+
+    return aText.localeCompare(bText);
+  });
+
+  rows.forEach((row) => tbody.appendChild(row));
+}
+
+const newForm = document.createElement('form');
+const body = document.querySelector('body');
+
+body.appendChild(newForm);
+newForm.setAttribute('class', 'new-employee-form');
+
+function makeLabel(newName) {
+  const label = document.createElement('label');
+  const input = document.createElement('input');
+
+  if (newName === 'Office') {
+    const select = document.createElement('select');
+    const offices = [
+      'Tokyo',
+      'Singapore',
+      'London',
+      'New York',
+      'Edinburgh',
+      'San Francisco',
+    ];
+
+    offices.forEach((office) => {
+      const option = document.createElement('option');
+
+      option.textContent = office;
+      select.appendChild(option);
+    });
+
+    newForm.appendChild(label);
+    label.textContent = `${newName}:`;
+    label.append(select);
+    select.setAttribute('name', 'name');
+    select.setAttribute('data-qa', newName.toLowerCase());
+
+    newForm.appendChild(label);
```

Comment:
This line is redundant because the label is already appended to `newForm` on line 69. You can remove this line to avoid appending the same label twice.